rust/operator/src/operations/graceful_shutdown.rs

```diff
@@ -0,0 +1,12 @@
+use stackable_hdfs_crd::MergedConfig;
+use stackable_operator::builder::PodBuilder;
+
+pub fn add_graceful_shutdown_config(
+    merged_config: &(dyn MergedConfig + Send + 'static),
+    pod_builder: &mut PodBuilder,
+) {
+    // This must be always set by the merge mechanism, as we provide a default value.
+    if let Some(graceful_shutdown_timeout) = merged_config.graceful_shutdown_timeout() {
+        pod_builder.termination_grace_period_seconds(graceful_shutdown_timeout.as_secs() as i64);
```

Comment:
This cast is dangerous. We basically half the capacity of a `u64` here. It's a bummer that `pod_builder.termination_grace_period_seconds()` takes an `i64`. We should maybe think about changing that (a negative amount of seconds doesn't even make sense).

Reply:
FWIW the `i64` type comes from [k8s-openapi](https://docs.rs/k8s-openapi/0.20.0/k8s_openapi/api/core/v1/struct.PodSpec.html#structfield.termination_grace_period_seconds) so we can't really get rid of it entirely, but agreed that I'd rather hide the ugliness in `PodBuilder`.

Reply:
Yeah. I feel like its worth to change that upstream. Otherwise we just move the problem around (from the operator into the `PodBuilder`). Even the [k8s-openapi](https://docs.rs/k8s-openapi/0.20.0/k8s_openapi/api/core/v1/struct.PodSpec.html#structfield.termination_grace_period_seconds) documentation states:

> Value must be non-negative integer.

So the `i64` makes even less sense (if it made any sense in the first place).

Reply:
Now using the new operator-rs version with the `PodBuilder::termination_grace_period` fn
@Techassi can we resolve this thread?

Reply:
Yes, we can resolve it for now.